chrome_agent/js/chrome_agent.js

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import JsFileSystem from 'axiom/fs/js/file_system';
+import SkeletonFileSystem from 'axiom/fs/stream/skeleton_file_system';
+import Transport from 'axiom/fs/stream/transport';
+import Channel from 'axiom/fs/stream/channel';
+import ExtensionStreams from 'axiom/fs/stream/extension_streams';
+
+/**
+ * Host filesystem with chrome handling executable for use with axiom.
+ *
+ * @constructor
+ */
+export var ChromeAgent = function() {
```

Comment:
I would leave the `ChromeAgent` constructor body empty, and move the current body to a new method (called `connect` maybe?) that return a promise.

The issue with putting this code inside the constructor is that 1) the caller doesn't know when the agent is ready and 2) the caller does not get notified of promise failures.